controllers/core/openstackclient_controller.go

```diff
@@ -0,0 +1,294 @@
+
+			if err := r.Status().Patch(ctx, instance, patch); err != nil && !k8s_errors.IsNotFound(err) {
+				util.LogErrorForObject(h, err, "Update status", instance)
+			}
+		}
+	}()
+
+	//
+	// Validate that keystoneAPI is up
+	//
+	keystoneAPI, err := keystonev1.GetKeystoneAPI(ctx, h, instance.Namespace, map[string]string{})
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				keystonev1.KeystoneAPIReadyCondition,
+				condition.ErrorReason,
+				condition.SeverityWarning,
+				corev1beta1.OpenStackClientReadyErrorMessage,
+				err.Error()))
+			r.Log.Info("KeystoneAPI not found!")
+			return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+		}
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenClientReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackClientReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	if !keystoneAPI.IsReady() {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenClientReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			corev1beta1.OpenStackClientKeystoneWaitingMessage))
+		r.Log.Info("KeystoneAPI not yet ready")
+		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+	}
+
+	_, configMapHash, err := configmap.GetConfigMapAndHashWithName(ctx, h, instance.Spec.OpenStackConfigMap, instance.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				corev1beta1.OpenClientReadyCondition,
+				condition.RequestedReason,
+				condition.SeverityInfo,
+				corev1beta1.OpenStackClientConfigMapWaitingMessage))
+			return ctrl.Result{RequeueAfter: time.Second * 10}, nil
+		}
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenClientReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackClientReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	_, secretHash, err := secret.GetSecret(ctx, h, instance.Spec.OpenStackConfigSecret, instance.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				corev1beta1.OpenClientReadyCondition,
+				condition.RequestedReason,
+				condition.SeverityInfo,
+				corev1beta1.OpenStackClientSecretWaitingMessage))
+			return ctrl.Result{RequeueAfter: time.Second * 10}, nil
+		}
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenClientReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackClientReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	instance.Status.Conditions.Set(condition.FalseCondition(
+		corev1beta1.OpenClientReadyCondition,
+		condition.RequestedReason,
+		condition.SeverityInfo,
+		corev1beta1.OpenStackClientInputReady))
```

Comment:
Isn't the input actually ready at this point?  Both the config map and the secret would have been found if we reach here.

Reply:
Ah, I see.  The condition is `OpenClientReadyCondition` and it's still not ready even if the input is ready.